### DISCUSSION TITLE
I keep getting a typescript 'unexpected token' error when using  a blockquote in an Astro MDX file and all my googling is coming up with no way of fixing it...

### DISCUSSION THREAD

**Discussion by PaulHaze** [mdx-js/mdx#2489] [General] [answered]
So I've recently started playing around MDX and am using the Starlight/Astro docs template.

Everything is working fine but when I follow a simple example from [https://mdxjs.com/docs/what-is-mdx/](https://mdxjs.com/docs/what-is-mdx/) of using a blockquote I constantly get a ts-plugin error:

<img width="500" alt="ts-plugin code error" src="https://github.com/mdx-js/mdx/assets/20569604/94383c5b-e0bd-4ba0-945e-36c168e25bb8">

The actual error:
```
Unexpected token. Did you mean `{'>'}` or `&gt;`?
```

When I actually view the page the blockquote appears just fine without issue:

<img width="479" alt="Screen Shot of working code" src="https://github.com/mdx-js/mdx/assets/20569604/851e5d75-1fc4-43a8-b1e8-bd08780c1ec7">

So I was wondering what is causing this issue, I've tried disabling eslint, disabling TS language services,  I've tried fixing it with both of the suggestions that pop up and that only solves it until I type something else:

<img width="381" alt="error suggestions" src="https://github.com/mdx-js/mdx/assets/20569604/0e874c59-b91a-4d9f-843e-92f72c0c5457">

If anyone could help me with this I would appreciate it. I can just ignore, but my OCD always like all problems to be 0!

**EDIT:** On further playing around, if I change the language mode at the bottom of VS Code from `MDX` to `markdown` the errors go away and the page still seems to work as expected. Am I losing anything by using markdown instead of MDX? It feels counterintuitive to set it to something that it isnt...

**Comment by wooorm** [comment]
/cc @remcohaszing seems mdx-analyzer / volar related.

This isn’t something in MDX itself.

**Comment by remcohaszing** [comment]
Thanks for reporting. This is definitely a bug in MDX analyzer and I managed to reproduce it.

---

> **EDIT:** On further playing around, if I change the language mode at the bottom of VS Code from `MDX` to `markdown` the errors go away and the page still seems to work as expected. Am I losing anything by using markdown instead of MDX? It feels counterintuitive to set it to something that it isnt...

MDX isn’t markdown. If you change the language of a file to something it isn’t, you get incorrect editor support. Since MDX looks a lot like markdown, it may seem ok at first, but you will run into issues.

**Reply by remcohaszing** [reply, marked as answer]
It’s fixed in https://github.com/mdx-js/mdx-analyzer/pull/448 and released in version [1.8.8](https://github.com/mdx-js/mdx-analyzer/releases/tag/vscode-mdx@1.8.8) of the VSCode MDX extension.

**Reply by PaulHaze** [reply]
Yep, I can confirm that has fixed it. That was a pretty rapid response!
